components/device-mgt/org.wso2.carbon.device.mgt.api/src/test/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/DeviceAgentServiceTest.java

```diff
@@ -0,0 +1,655 @@
+import org.wso2.carbon.device.mgt.common.DeviceIdentifier;
+import org.wso2.carbon.device.mgt.common.DeviceManagementException;
+import org.wso2.carbon.device.mgt.common.EnrolmentInfo;
+import org.wso2.carbon.device.mgt.common.authorization.DeviceAccessAuthorizationException;
+import org.wso2.carbon.device.mgt.common.authorization.DeviceAccessAuthorizationService;
+import org.wso2.carbon.device.mgt.core.authorization.DeviceAccessAuthorizationServiceImpl;
+import org.wso2.carbon.device.mgt.core.service.DeviceManagementProviderService;
+import org.wso2.carbon.device.mgt.core.service.DeviceManagementProviderServiceImpl;
+import org.wso2.carbon.device.mgt.jaxrs.service.api.DeviceAgentService;
+import org.wso2.carbon.device.mgt.jaxrs.service.api.admin.DeviceTypeManagementAdminService;
+import org.wso2.carbon.device.mgt.jaxrs.service.impl.util.DeviceMgtAPITestHelper;
+import org.wso2.carbon.device.mgt.jaxrs.util.DeviceMgtAPIUtils;
+import org.wso2.carbon.event.stream.stub.EventStreamAdminServiceStub;
+import org.wso2.carbon.event.stream.stub.types.EventStreamAttributeDto;
+import org.wso2.carbon.event.stream.stub.types.EventStreamDefinitionDto;
+import org.wso2.carbon.identity.jwt.client.extension.exception.JWTClientException;
+import org.wso2.carbon.user.api.UserStoreException;
+import org.wso2.carbon.utils.CarbonUtils;
+
+import javax.cache.CacheManager;
+import javax.ws.rs.core.Response;
+import java.rmi.RemoteException;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.mockito.MockitoAnnotations.initMocks;
+
+/**
+ * This class holds the unit tests for the class {@link DeviceAgentServiceImpl}
+ */
+@PowerMockIgnore("javax.ws.rs.*")
+@SuppressStaticInitializationFor({"org.wso2.carbon.device.mgt.jaxrs.util.DeviceMgtAPIUtils",
+        "org.wso2.carbon.context.CarbonContext", "org.wso2.carbon.context.internal.CarbonContextDataHolder"})
+@PrepareForTest({DeviceMgtAPIUtils.class, DeviceManagementProviderService.class,
+        DeviceAccessAuthorizationService.class, EventStreamAdminServiceStub.class, PrivilegedCarbonContext.class,
+        CarbonContext.class, CarbonUtils.class})
+public class DeviceAgentServiceTest {
+
+    private static final Log log = LogFactory.getLog(DeviceTypeManagementAdminService.class);
+    private DeviceManagementProviderService deviceManagementProviderService;
+    private DeviceAgentService deviceAgentService;
+    private EventStreamAdminServiceStub eventStreamAdminServiceStub;
+    private PrivilegedCarbonContext privilegedCarbonContext;
+    private CarbonContext carbonContext;
+    private CacheManager cacheManager;
+    private DeviceAccessAuthorizationService deviceAccessAuthorizationService;
+    private static final String TEST_DEVICE_TYPE = "TEST-DEVICE-TYPE";
+    private static final String TEST_DEVICE_IDENTIFIER = "11222334455";
+    private static final String AUTHENTICATED_USER = "admin";
+    private static final String TENANT_DOMAIN = "carbon.super";
```

Comment:
Shall we use super-tenant domain name from MutiTenantConstants.

Reply:
This is fixed in the commit 
https://github.com/wso2/carbon-device-mgt/pull/1082/commits/1a995dde138fc934f14bf626ea0506061f04f39e